portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -4878,10 +4973,6 @@ public User updatePassword(
 
 		if (!silentUpdate) {
 			user.setPasswordModified(false);
-
-			sendPasswordNotification(
-				user, user.getCompanyId(), password1, null, null, null, null,
-				null, serviceContext);
```

Comment:
Moving the `sendPasswordNotification` out from here will require third parties to reimplement their solutions I guess, so it would have quite a meaningful impact. I would avoid that if possible.

Reply:
I understand, but it's hard to get around the problem in the current way the code is implemented. It validates the login along with sending the email, so I thought of splitting the whole method into two.